src/jmclient/wallet.py

```diff
@@ -2644,6 +2800,43 @@ def _get_bip32_export_path(self, mixdepth=None, address_type=None):
         path = super()._get_bip32_export_path(mixdepth, address_type)
         return path
 
+    def _get_key_from_path(self, path,
+            validate_cache: bool = False):
+        raise WalletError("Cannot get a private key from a watch-only wallet")
+
+    def _get_keypair_from_path(self, path,
+            validate_cache: bool = False):
+        raise WalletError("Cannot get a private key from a watch-only wallet")
```

Comment:
Same error message is raised at multiple places, would be better to create custom exception derived from `WalletError` to avoid duplications.

Reply:
#1610